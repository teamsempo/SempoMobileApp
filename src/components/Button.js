import React from 'react';
import { View, Text, StyleSheet, TouchableHighlight } from 'react-native';

export default class Button extends React.Component {
    constructor() {
        super();
    }

    render() {
        let { outline } = this.props;
        return(
            <View>
                <TouchableHighlight underlayColor='#34b0b3' style={[styles.submitButton, {backgroundColor: outline ? 'transparent' : '#2D9EA0', borderWidth: outline ? 2 : 0, borderColor: outline ? '#2D9EA0' : null}]} onPress={this.props.onPress} accessibilityLabel={this.props.buttonText}>
                    <Text style={[styles.chargeButtonText, {color: outline ? '#2D9EA0' : '#FFF'}]}>{this.props.buttonText.toUpperCase()}</Text>
                </TouchableHighlight>
            </View>
        )
    }
}

Button.defaultProps = {
    outline: false,
};

const styles = StyleSheet.create({
    chargeButtonText: {
        fontSize: 12,
        textAlign: 'center',
        fontWeight: '500',
        letterSpacing: 1,
    },
    submitButton: {
        display: 'flex',
        alignItems: 'center',
        justifyContent: 'center',
        paddingTop: 12,
        paddingBottom: 12,
        paddingLeft: 24,
        paddingRight: 24,
        borderWidth: 0,
        margin: 5,
        borderRadius: 2,
    }
});