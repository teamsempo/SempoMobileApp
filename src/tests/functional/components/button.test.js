import React from 'react';
import { shallow } from 'enzyme';
import Button from '../../../components/Button.js';

describe('Button', () => {
    describe('Rendering', () => {
        it('should match to snapshot', () => {
            const mockCallBack = jest.fn();
            const component = shallow(<Button buttonText="SIGN IN" onPress={mockCallBack}/>);
            expect(component).toMatchSnapshot()
            component.find('ForwardRef').simulate('press');
            expect(mockCallBack).toHaveBeenCalled();
        });
    });
});
