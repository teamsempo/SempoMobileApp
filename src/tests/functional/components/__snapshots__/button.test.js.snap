// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Rendering should match to snapshot 1`] = `
<View>
  <ForwardRef
    accessibilityLabel="SIGN IN"
    onPress={[MockFunction]}
    style={
      Array [
        Object {
          "alignItems": "center",
          "borderRadius": 2,
          "borderWidth": 0,
          "display": "flex",
          "justifyContent": "center",
          "margin": 5,
          "paddingBottom": 12,
          "paddingLeft": 24,
          "paddingRight": 24,
          "paddingTop": 12,
        },
        Object {
          "backgroundColor": "#2D9EA0",
          "borderColor": null,
          "borderWidth": 0,
        },
      ]
    }
    underlayColor="#34b0b3"
  >
    <Text
      style={
        Array [
          Object {
            "fontSize": 12,
            "fontWeight": "500",
            "letterSpacing": 1,
            "textAlign": "center",
          },
          Object {
            "color": "#FFF",
          },
        ]
      }
    >
      SIGN IN
    </Text>
  </ForwardRef>
</View>
`;
